### PR TITLE
VRBRAIN: updated AP_HAL_VRBRAIN, added new boards and updated makefiles and ROMFS

### PR DESCRIPTION
**Pull request by LukeMike**
- Updated AP_HAL_VRBRAIN
- updated customizations of some libraries
- updated environment and targets definition
- updated VRBRAIN makefiles and ROMFS

### CODE REVIEW

File: libraries/AP_BattMonitor/AP_BattMonitor_Analog.h [hunk 1]
```diff
@@ -24,27 +24,21 @@
...
  # define AP_BATT_VOLT_PIN                  10
  # define AP_BATT_CURR_PIN                  1
  # define AP_BATT_VOLTDIVIDER_DEFAULT       10.1f
  # define AP_BATT_CURR_AMP_PERVOLT_DEFAULT  17.0f
-#endif
-#elif CONFIG_HAL_BOARD == HAL_BOARD_VRBRAIN && defined(CONFIG_ARCH_BOARD_VRHERO_V10)
- # define AP_BATT_VOLT_PIN                  100
- # define AP_BATT_CURR_PIN                  101
- # define AP_BATT_VOLTDIVIDER_DEFAULT       1.1f
  # define AP_BATT_CURR_AMP_PERVOLT_DEFAULT  17.0f
```
Comment (OXINARF) [inline libraries/AP_BattMonitor/AP_BattMonitor_Analog.h]: This line was forgotten, should be removed as well.

File: libraries/AP_BoardConfig/AP_BoardConfig.cpp [hunk 2]
```diff
@@ -168,13 +177,17 @@ void AP_BoardConfig::init()
             hal.console->printf("RCOutput: unable to setup AUX PWM with BRD_PWM_COUNT %u\n", mode_parm);
         }   
         close(fd);
+#if CONFIG_HAL_BOARD == HAL_BOARD_PX4
         if (mode_table[i].num_gpios < 2) {
             // reduce change of config mistake where relay and PWM interfere
             AP_Param::set_default_by_name("RELAY_PIN", -1);
             AP_Param::set_default_by_name("RELAY_PIN2", -1);
         }
+#endif
```
Comment (OXINARF) [inline libraries/AP_BoardConfig/AP_BoardConfig.cpp]: Why doesn't this apply to VRBRAIN boards?

File: mk/vrbrain_targets.mk [hunk 3]
```diff
@@ -1,467 +1,264 @@
...
+ifneq ($(wildcard $(SKETCHBOOK)/../uavcan),)
+$(warning *** You have an old uavcan tree - see http://dev.ardupilot.com/wiki/git-submodules/)
+endif
+
+NUTTX_GIT_VERSION ?= $(shell cd $(NUTTX_SRC) && git rev-parse HEAD | cut -c1-8)
+PX4_GIT_VERSION   ?= $(shell cd $(VRBRAIN_ROOT) && git rev-parse HEAD | cut -c1-8)
+
+EXTRAFLAGS += -DNUTTX_GIT_VERSION="\"$(NUTTX_GIT_VERSION)\""
+EXTRAFLAGS += -DPX4_GIT_VERSION="\"$(PX4_GIT_VERSION)\""
+EXTRAFLAGS += -DUAVCAN=1
```
Comment (OXINARF) [inline mk/vrbrain_targets.mk]: I'm not sure where this define is used but am I mistaken that VRBRAIN doesn't have UAVCAN?

File: mk/VRBRAIN/Tools/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+These tools are from the ros project. You can find the master sources
+here:
+
+http://github.com/ros/gencpp
+http://github.com/ros/genmsg
```
Comment (OXINARF) [inline mk/VRBRAIN/Tools/README.md]: Why import these here? This is already part of PX4Firmware as a submodule...

File: mk/VRBRAIN/config_vrubrain-v52_APM.mk [hunk 5]
```diff
@@ -1,172 +1,10 @@
...
+#
+# Makefile for the VRUBRAIN 5.2 APM configuration
+#
+include $(SKETCHBOOK)/mk/VRBRAIN/vrbrain_common.mk
+
+#MODULES		+= drivers/mpu9250
+MODULES		+= drivers/boards/vrubrain-v52
+MODULES		+= drivers/pwm_input
+#MODULES         += modules/uavcan
+MODULES         += lib/mathlib
```
Comment (OXINARF) [inline mk/VRBRAIN/config_vrubrain-v52_APM.mk]: drivers/pwm_input and lib/mathlib seem to be common to all of them so they could go to vrbrain_common.mk too.

**Comment by OXINARF** [comment]
@LukeMike I made some comments (not sure why one of them appears as it was in an outdated diff but it isn't). I haven't done a thorough review, it was just looking superficially. Great to see this coming, supporting a board in master is always a good thing!

File: libraries/AP_Notify/ExternalLED.h [hunk 6]
```diff
@@ -23,6 +23,9 @@
 
 #include "NotifyDevice.h"
 
+#define HIGH 1
+#define LOW 0
```
Comment (OXINARF) [inline libraries/AP_Notify/ExternalLED.h]: What is this used for? Did I miss something?

Comment (LukeMike) [inline libraries/AP_Notify/ExternalLED.h]: Into ExternalLED.cpp HAL_GPIO_LED_OFF, defined into AP_HAL/AP_HAL_Boards.h, don't see the definition of HIGH and LOW. Is the same thing of AP_BoardLED.cpp (HIGH and LOW are defined into AP_BoardLED.h)

File: mk/VRBRAIN/config_vrubrain-v52_APM.mk (hunk 5 above)
Comment (LukeMike) [inline mk/VRBRAIN/config_vrubrain-v52_APM.mk]: It's true. I'm just trying to keep a certain parallelism with PX4 ... maybe in the future I can make our boards as SUBTYPE of PX4.

File: mk/VRBRAIN/Tools/README.md (hunk 4 above)
Comment (LukeMike) [inline mk/VRBRAIN/Tools/README.md]: Is the same of PX4. Maybe we can remove these Tools both in VRBRAIN and PX4

File: libraries/AP_BattMonitor/AP_BattMonitor_Analog.h (hunk 1 above)
Comment (LukeMike) [inline libraries/AP_BattMonitor/AP_BattMonitor_Analog.h]: Ok. VR Hero board can be removed

File: mk/VRBRAIN/Tools/README.md (hunk 4 above)
Comment (OXINARF) [inline mk/VRBRAIN/Tools/README.md]: I didn't notice that was there. It doesn't seem to make sense to have it in ArduPilot and PX4Firmware but I'm not sure if it is still needed or not. Even if it is needed to be in ArduPilot you can use the one already in the PX4 folder, no need to have this duplicated in ArduPilot (we would end up with 3 versions).

File: libraries/AP_Notify/ExternalLED.h (hunk 6 above)
Comment (OXINARF) [inline libraries/AP_Notify/ExternalLED.h]: Ok. HIGH/LOW are really bad defines but they are there so I think it's fine to add these ones.

**Comment by lucasdemarchi** [comment]
I'm a little bit confused why this isn't a board subtype of PX4.

**Comment by OXINARF** [comment]
@lucasdemarchi These aren't PX4 boards. They have many similarities and share lots of code though.

**Comment by tridge** [comment]
@OXINARF in the future I would prefer that VRBrain be done as a PX4 subtype, as it would same quite a bit of maintainence work I think
It doesn't have to be in this round of merges, but it would be a nice goal

**Comment by tridge** [comment]
trying to rebase this on master I am getting quite a few conflicts. Luke, could you rebase it please?

**Comment by OXINARF** [comment]
@tridge I agree that if we can make both use the same HAL it would be great, but that shouldn't stop the merge of this.
And I have to say it is really a bummer that PX4 name is used for everything, from boards to firmware.

**Comment by LukeMike** [comment]
@tridge I rebased my branch in master. I see that all checks have passed. Now everything should be ok.

**Comment by tridge** [comment]
I have pushed this to master, thanks!

**Comment by OXINARF** [comment]
@tridge Don't forget https://github.com/ArduPilot/PX4Firmware/pull/58 otherwise I think this won't work.

**Pull request closed without merging**